presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -1154,6 +1159,82 @@ public void testShowCreateTable()
         assertEquals(getOnlyElement(actualResult.getOnlyColumnAsSet()), createTableSql);
     }
 
+    @Test
+    public void testPathHiddenColumn()
+            throws Exception
+    {
+        for (HiveStorageFormat storageFormat : HiveStorageFormat.values()) {
+            selectPath(storageFormat);
+        }
+    }
+
+    private void selectPath(HiveStorageFormat storageFormat)
+    {
+        @Language("SQL") String createTable = "" +
+                "CREATE TABLE test_path (" +
+                "  col0 bigint" +
+                ", col1 integer" +
+                ") " +
+                "WITH (" +
+                "format = '" + storageFormat + "'," +
+                "partitioned_by = ARRAY['col1']" +
+                ") ";
+        assertUpdate(createTable);
+        assertTrue(queryRunner.tableExists(getSession(), "test_path"));
+
+        TableMetadata tableMetadata = getTableMetadata(catalog, TPCH_SCHEMA, "test_path");
+        assertEquals(tableMetadata.getMetadata().getProperties().get(STORAGE_FORMAT_PROPERTY), storageFormat);
+
+        List<String> columnNames = ImmutableList.of("col0", "col1", HiveColumnHandle.PATH_COLUMN_NAME);
+        List<ColumnMetadata> columnMetadatas = tableMetadata.getColumns();
+        assertEquals(columnMetadatas.size(), columnNames.size());
+        for (int i = 0; i < columnMetadatas.size(); i++) {
+            ColumnMetadata columnMetadata = columnMetadatas.get(i);
+            assertEquals(columnMetadata.getName(), columnNames.get(i));
+            if (columnMetadata.getName().equals(HiveColumnHandle.PATH_COLUMN_NAME)) {
+                // $path should be hidden column
+                assertTrue(columnMetadata.isHidden());
+            }
+        }
+
+        int partitionCount = 3;
+        int valueCount = 123;
+        List<String> values = new ArrayList<>();
+        for (int i = 0; i < valueCount; i++) {
+            values.add(format("(%d, %d)", i, i % partitionCount));
+        }
+
+        @Language("SQL") String insertValues = "INSERT INTO test_path VALUES " + Joiner.on(",").join(values);
```

Comment:
I think this test is overly complicated. You don't need to create table and then insert into the table. Use `CREATE TABLE AS SELECT` instead. Additionally, you don't need to dynamically generate the columns. e.g. manually write a values of 3 partitions and 8 rows is probably enough.